docs/rfcs/0004-suggested-excluded-apps-tech-spec.md

```diff
@@ -0,0 +1,142 @@
+-   (2-5 points) Ensure accessibility is up to date
+-   (1 point) Add metrics for how many suggested (and non-suggested) apps are excluded\
+
+### Day 2
+
+These are important components that we are committed to delivering, but are outside the scope of this epic. They are items that will be done as soon as the epic is closed, but do not block the feature on shipping.
+
+-   Engineering metrics
+    -   Metrics will be created for a successful deep linking, as well as an erroneous deep linking (the app cannot figure out how to resolve the deep link).
+    -   Add performance metrics around load time of the Excluded Apps screen.
+    -   Create a metric for when the daily job was successful, and another one for when it fails.
+    -   A metric should be recorded when the notification is sent
+
+Other options considered
+------------------------
+
+Design and research went through several iterations, so the [designs](https://www.figma.com/file/UZYzma7hlcfE5ke3z8jGbN/App-exclusions-suggestions?type=design&node-id=196-6366&mode=design&t=RL1hdfBQLMS1rKVa-0) were set before this tech spec was written. (See [completed tickets in the epic](https://mozilla-hub.atlassian.net/browse/VPN-4412) for more background.)
+
+Other options considered included dynamically serving the suggested lists from addons or our repo or Guardian, each of which added complexity for hosting/validating. With the rarity of the list being updated (and lack of time-sensitivity when it does), this draft bundles the lists with the app. Hosting the list in Guardian was difficult (which is a private repo, so the list would need to be hosted elsewhere to allow public suggestions).
+
+### Other options considered for suggested app files
+
+The goals for the file format:
+-   Easy to read by humans
+-   Easy for humans to add to, no matter their level of technical proficiency
+-   Easy for us to parse in the app
+-   Allowing comments in the file. While we still could structure a text file to allow comments, this became less important when the plan became "one file per platform, one app per line".
+
+Initially, there was one additional goal (no longer consideration):
+-   To have one file for all platforms, so that we'd remember to add a specific app to all platforms at once. With a conversation around potentially wanting different suggested app categories on mobile, this seems unimportant.
+
+Other options considered included CSV, YAML, and JSON. While YAML could be useful (as it's easy for less-technical people to read), we don't have a good YAML parser in the app yet. We could add a library, of course. However, we'd need to write our own document format test, because the YAML file (or CSV) would need to be structured in a specific way. While we already have a JSON parser that we've custom wrote that could be used for this validation, a simple list is better than JSON for the first two goals of the file format.
```

Comment:
We did not custom write it. We just wrote the schema, but [JSON schema](https://json-schema.org/) is not a tool we created. 

I believe it would probably be more reliable and easier to use JSON here. Because Qt already has tools to parse JSON and it is trivial to write a simple schema and validate it.

Reply:
I don't see how JSON is more reliable and easier than a file with one entry per line, especially if we want to make it simple for less-technical people to contribute to this. Can you explain?

Reply:
Technically I believe it would be easier because the tools for parsing and validating are already there. 

With regards to less technical people contributing, we could go with an extremely minimal JSON format here. Say we decide to keep almost exactly what is currently being proposed for `txt` files and have `android.json`, `windows.json` etc. The JSON format could look like:

```json
[
   "my.app.identifier",
   "my.other.app.identifier"
]
```

And so on, which just adds the quotes, commas and brackets more than what the `txt` . However, with the JSON schema we would be able to validate the format and create regular expression to validate the format of each identifier per platform without much overhead e.g. the JSON schema for `android.json` would look something like:

```json
{
  "type": "array",
  "items": {
    "type": "string",
    "pattern": "^[a-zA-Z][a-zA-Z0-9_]*(\\.[a-zA-Z][a-zA-Z0-9_]*)*$"
  }
}
```

Reply:
Thanks for all this. I'll update the doc today or tomorrow.

Reply:
Okay, I think I've come around. I'll update this doc today or tomorrow.